src/qx/Simulator.cpp

```diff
@@ -13,67 +13,59 @@
 
 namespace qx {
 
-std::optional<SimulationError> Simulator::set(std::string const &filePath) {
-    auto analyzer = cqasm::v1x::default_analyzer("1.2");
-
-    auto analysisResult = analyzer.analyze(filePath);
-    if (!analysisResult.errors.empty()) {
-        program.reset();
-        std::stringstream error;
-        error << "Cannot parse and analyze file " << filePath << ":";
-
-        std::for_each(analysisResult.errors.begin(), analysisResult.errors.end(), [&error](auto const& s) { error << "\n" << s; });
-
-        return SimulationError{error.str()};
-    }
+using AnalysisResult = cqasm::v1x::analyzer::AnalysisResult;
+using Program = cqasm::v1x::ast::One<cqasm::v1x::semantic::Program>;
 
-    program = analysisResult.root;
-    assert(!program.empty());
-
-    if (!program->error_model.empty() &&
-        program->error_model->name != "depolarizing_channel") {
-        program.reset();
-        return SimulationError{"Unknown error model: " + program->error_model->name};
-    }
+namespace {
+AnalysisResult parseCQASMV1File(std::string const &filePath) {
+    auto analyzer = cqasm::v1x::default_analyzer("1.2");
 
-    return std::nullopt;
+    return analyzer.analyze(filePath);
 }
 
-std::optional<SimulationError> Simulator::setString(std::string const &s) {
+AnalysisResult parseCQASMV1String(std::string const &s) {
     auto analyzer = cqasm::v1x::default_analyzer("1.2");
 
-    auto analysisResult = analyzer.analyze_string(s);
+    return analyzer.analyze_string(s);
+}
+
+std::variant<Program, SimulationError> getProgramOrError(AnalysisResult const& analysisResult) {
     if (!analysisResult.errors.empty()) {
-        program.reset();
         std::stringstream error;
-        error << "Cannot parse and analyze string " << s << ": \n";
+        error << "Cannot parse and analyze cQASM: \n";
 
         std::for_each(analysisResult.errors.begin(), analysisResult.errors.end(), [&error](auto const& x) { error << x << "\n"; });
 
         return SimulationError{error.str()};
     }
 
-    program = analysisResult.root;
+    auto program = analysisResult.root;
     assert(!program.empty());
 
     if (!program->error_model.empty() &&
         program->error_model->name != "depolarizing_channel") {
         std::cerr << "Unknown error model: " << program->error_model->name
-                  << std::endl;
-        program.reset();
+                << std::endl;
         return SimulationError{"Unknown error model: " + program->error_model->name};
     }
 
-    return std::nullopt;
+    return program;
 }
-
+    
 std::variant<SimulationResult, SimulationError>
-Simulator::execute(std::size_t iterations,
-                   std::optional<std::uint_fast64_t> seed) const {
-    if (program.empty()) {
-        return SimulationError{"No circuit successfully loaded, call set(...) first"};
+execute(AnalysisResult analysisResult,
+                std::size_t iterations,
```

Comment:
Sorry for being too picky here. These two lines seem to be aligned to some old code.

Reply:
Fixed.